Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
+   {
+       this->Delete();
+       return;
+   }
+   current_step_++;
+   if(!(current_step_ < movement_.size()) && !reached_target )
+   {
+       current_step_ = 0;
+       VDir temp = movement_[0];
+       movement_.clear();
+       reached_target = true;
+       movement_.push_back(temp);
+   }
+   if(!(current_step_<movement_.size()) && reached_target )
+   {
+   current_step_ = 0;
+   }   
+}
+
+bool Projectile::CheckPassable()
+{
+    id_ptr_on<Projectile> p = this->GetId();
```

Comment:
Every object in Griefly has `transparent_` variable. It can be accessed via `IsTransparent` function. That should be enough for laser projectlies. I do not see the need in a new variable here.